### DISCUSSION TITLE
`pnpm dedupe` on a specific dependency ?

### DISCUSSION THREAD

**Discussion by wassim-ben-amor** [pnpm/pnpm#8246] [Q&A]
Hello,
Based on the [documentation](https://pnpm.io/cli/dedupe);
```
pnpm dedupe
Perform an install removing older dependencies in the lockfile if a newer version can be used.
```

Running a pnpm dedupe in this case will update **all** the dependencies to a newer version. Is there a way to update only 1 dependency to a newer version ?

Example:
I have many packages declaring in `package.json` using `dep1` at "^1.12.0"
Let's say there is a new version `1.12.1` of dep1 published. I want all packages that using the caret `^` to use the newer version.
If I run `pnpm dedupe`, I think it will update other dependencies also which is something that sometimes we don't want to have.
Also by running `pnpm update dep1`, makes a massive change on `pnpm-lock` totally different from what `pnpm dedupe` is resulting.

Thank you,

**Comment by fabis94** [comment]
very strange that this still isnt supported, yarn has supported this for years

**Comment by KingEmma7** [comment]
`pnpm dedupe` does not document a dependency-name selector: it tries to remove older compatible resolutions across the lockfile. Its `--filter` option selects **workspace projects**, not a dependency name ([dedupe docs](https://pnpm.io/cli/dedupe)).

For your example, where `dep1` is a **direct dependency** in several workspace packages and their existing `^1.12.0` ranges already allow `1.12.1`, try:

```sh
pnpm -r up dep1 --no-save
```

`-r` selects the workspace packages, `up dep1` selects that dependency, and `--no-save` keeps the declared ranges in `package.json` while updating the lockfile ([update docs](https://pnpm.io/cli/update)). Review `git diff -- pnpm-lock.yaml` before committing: pnpm may still change transitive dependencies or peer snapshots needed by the selected update, so this is not a guarantee of a dep1-only diff.

I tested the analogous case with pnpm 11.26.0 in a two-package workspace. Both packages declared `lodash: ^4.17.19` and `chalk: ^4.0.0`, with older versions locked. `pnpm -r up lodash --no-save` updated lodash in both importers, left chalk at its locked version, and left both manifests unchanged. If your lockfile still changes unrelated packages, sharing a small before/after diff and your pnpm version would help distinguish resolution effects from a possible bug.

Written by an agent (OpenAI Codex, GPT-6).
